### PR TITLE
feat: add transpile_sql and execute_last_sql for demo 

### PR DESCRIPTION
**Pull request by jitingxu1**
1. add transpile_sql and execute_last_sql for demo.
2. Rename code attachment attribute `language` to `dialect`. we need to know the dialect of the executed sql for the `transpile_sql`, IMO, it would be better to change it to dialect, even though the dialect could be inferred from the `con`.

### CODE REVIEW

File: src/ibis_birdbrain/bot.py [hunk 1]
```diff
@@ -172,12 +175,42 @@ def respond(self, messages: Messages) -> Message:
         """Respond to the messages."""
         ...
 
-    # TODO: for demo
-    def translate_sql(self, sql: str, dialect_to: str, dialect_from: str) -> str:
+    def transpile_sql(self, sql: str, dialect_from: str, dialect_to: str) -> str:
         """Translate SQL from one dialect to another."""
-        ...
+        
+        return sg.transpile(sql, read=dialect_from, write=dialect_to, identity=False)[0]
```
Comment (lostmygithubaccount) [inline src/ibis_birdbrain/bot.py]: should probably use `pretty=True`

File: src/ibis_birdbrain/bot.py [hunk 2]
```diff
@@ -172,12 +175,42 @@ def respond(self, messages: Messages) -> Message:
...
+                if isinstance(message.attachments[attachment], CodeAttachment):
+                    sql_attachment = message.attachments[attachment]
+                if isinstance(message.attachments[attachment], TableAttachment):
+                    table_attachment = message.attachments[attachment]
+            if sql_attachment and table_attachment:
+                # Only execute last succefully executed sql query who has a table attachment
+                break
+            else:
+                sql_attachment = None
+                table_attachment = None
```
Comment (lostmygithubaccount) [inline src/ibis_birdbrain/bot.py]: might be worth implementing better methods on attachments for getting ones of certain types, and refactoring this + elsewhere

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
the intent of `CodeAttachment` was to have SQL and Python code -- could we instead just add `dialect` as an attribute that's used if `language == "sql"`? or otherwise rename to SQLAttachment, and introduce PythonAttachment later

we want to be able to show the SQL and corresponding Ibis code side-by-side as soon as possible

File: src/ibis_birdbrain/bot.py (hunk 2 above)
Comment (jitingxu1) [reply src/ibis_birdbrain/bot.py]: Yes, I thought the same, looks like every message will only have at most one attachment from the same type, we could use the attachment type as the key, we could get the attachment by type. How do you think? @lostmygithubaccount

Comment (lostmygithubaccount) [reply src/ibis_birdbrain/bot.py]: sorry for the slow response here. in theory, there could be multiple attachments of the same type but best to keep it simple

**Review by lostmygithubaccount — APPROVED** [review]
thanks!

**Pull request merged**